src/transaction/Transaction.js

```diff
@@ -851,6 +851,86 @@ export default class Transaction extends Executable {
         return this;
     }
 
+    /**
+     * This method removes a signature from the transaction based on the public key or signature provided.
+     *
+     * @param {PublicKey} publicKey - The public key associated with the signature to remove.
+     * @param {Uint8Array} [signature] - The signature to remove. If omitted, all signatures for the given public key will be removed.
+     * @returns {Uint8Array[]} The removed signatures.
+     */
+    removeSignature(publicKey, signature) {
+        // If the transaction isn't frozen, freeze it.
+        if (!this.isFrozen()) {
+            this.freeze();
+        }
+
+        const publicKeyData = publicKey.toBytesRaw();
+        const publicKeyHex = hex.encode(publicKeyData);
+
+        if (!this._signerPublicKeys.has(publicKeyHex)) {
+            throw new Error("The public key has not signed this transaction");
+        }
+
+        /** @type {Uint8Array[]} */
+        const removedSignatures = [];
+
+        // Iterate over the signed transactions and remove matching signatures
+        for (const transaction of this._signedTransactions.list) {
+            const removedSignaturesFromTransaction =
+                this._removeSignaturesFromTransaction(
+                    transaction,
+                    publicKeyHex,
+                    signature,
+                );
+
+            removedSignatures.push(...removedSignaturesFromTransaction);
+        }
+
+        // Remove the public key from internal tracking if no signatures remain
+        this._signerPublicKeys.delete(publicKeyHex);
+        this._publicKeys = this._publicKeys.filter(
+            (key) => !key.equals(publicKey),
+        );
+
+        // Update transaction signers array
+        this._transactionSigners.pop();
+
+        return removedSignatures;
+    }
+
+    /**
+     * This method clears all signatures from the transaction and returns them in a specific format.
+     *
+     * It will call collectSignatures to get the removed signatures, then clear all signatures
+     * from the internal tracking.
+     *
+     * @returns {{ [transactionId: string]: { [nodeAccountId: string]: Uint8Array[] } }} The removed signatures in the specified format.
```

Comment:
The data structure for this return type should probably be `{ [publicKey: PublicKey]: Uint8Array[] }` to be consistent with the `removeSignature` function, I would think. Or is that not preferable?

Reply:
I'll rename the function to removeAllSignatures for consistency. I also agree that it would be simpler and more consistent to return the publicKey associated with the signatures. Since each publicKey can only have one associated signature, the return type won't need to be an array. This approach aligns better with the logic in the removeSignature function.